### PR TITLE
fix(release): decouple title from rendered text in TextItem (CX-16559)

### PR DESCRIPTION
**Pull request by cormacmchale1**
This will allow users to optionally avoid the browser generated tooltip from title.

## Description

This will allow users to optionally avoid the browser generated tooltip from title. By providing a new variable to use instead of title as the display text. If title is only used. Then the legacy functionalty will reamin and it will be used as display text.

## Motivation and Context

Prevent multiple tooltips in the case where a tooltip has been added to a task-item.

## How Has This Been Tested?

Will not affect anything external to the component, or any pre-exsisting functionality on the component.
Run and verified in the web-component sandbox.
Run against the suite of tests exisiting for the component, with tests added to cover my change.

## Screenshots:
**Before** (If applicable):

Functionality is preserved in this case. This changes allows any consumer the ability to remove the broswer tooltip but retain the ability to display the visible title on screen.

**After**:

Screenshot 2024-11-13 at 09 04 54
Screenshot 2024-11-13 at 09 05 04
Screenshot 2024-11-13 at 09 05 11

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: web-components/src/components/taskitem/TaskItem.ts [hunk 1]
```diff
@@ -31,6 +30,37 @@ export namespace TaskItem {
...
+    @property({ type: String, reflect: true})
+    get itemTitle() {
+      return this.displayTitle;
+    }
+    set itemTitle(itemTitle: string) {
+      this.updateTitle(itemTitle);
+    }
+
+    private updateTitle(title: string) {
+      if (!title) {
```
Comment (cormacmchale1) [inline web-components/src/components/taskitem/TaskItem.ts:56]: This logic covers the cases of passing blank strings for either variable. The test component is setup in this way, seems like a legitimate case to cover

File: web-components/src/components/taskitem/TaskItem.test.ts [hunk 2]
```diff
@@ -18,8 +19,9 @@ const fixtureFactory = async (
   return await fixture(html`
     <md-task-item
       mediaType="${mediaType}"
-      title="${title}"
       popovertitle="${popovertitle}"
+      itemTitle="${itemTitle}"
+      title="${title}"
```
Comment (cormacmchale1) [inline web-components/src/components/taskitem/TaskItem.test.ts:24]: Just to make sure the ordering of these don't break the tests with new logic

File: web-components/src/components/taskitem/TaskItem.ts [hunk 3]
```diff
@@ -30,6 +29,43 @@ export namespace TaskItem {
...
+      this.itemTitleValue = itemTitle;
+      this.updateTitle();
+    }
+
+    private updateTitle() {
+      if(!this.itemTitleValue){
+      this.displayTitle = this.titleValue;
+      }
+      else{
+        this.displayTitle = this.itemTitleValue;
```
Comment (cormacmchale1) [inline web-components/src/components/taskitem/TaskItem.ts:64]: We just prioritise itemTitle for display text. But will use title if nothing is provided.

Comment (NotNestor) [reply web-components/src/components/taskitem/TaskItem.ts:64]: removing title is a breaking API change. this should be marked as deprecated and left

File: web-components/src/components/taskitem/TaskItem.ts [hunk 4]
```diff
@@ -30,6 +29,43 @@ export namespace TaskItem {
...
+    }
+    set title(title: string) {
+      this.titleValue = title;
+      this.updateTitle();
+    }
+    /**
+    * The visible text for the Title.
+    */
+    @property({ type: String, reflect: true, attribute: "itemtitle" })
+    get itemTitle() {
```
Comment (NotNestor) [inline web-components/src/components/taskitem/TaskItem.ts:51]: update the attribute to item-title

Comment (NotNestor) [reply web-components/src/components/taskitem/TaskItem.ts:51]: when this is item-title you can use the attribute item-title="title" or .itemTitle="title"

File: web-components/src/components/taskitem/TaskItem.ts (hunk 3 above)
Comment (cormacmchale1) [reply web-components/src/components/taskitem/TaskItem.ts:64]: This doesn't remove it though, sorry if I'm not following, any legacy usage is preserved in this PR.

File: web-components/src/components/taskitem/TaskItem.ts [hunk 5]
```diff
@@ -30,6 +29,43 @@ export namespace TaskItem {
     @property({ type: String }) customAriaLabel = "";
     @property({ type: String }) iconSrc = "";
     @property({ type: String }) tabIndexForContainer = "0";
+    @property({ type: String }) titleValue = "";
```
Comment (cormacmchale1) [inline web-components/src/components/taskitem/TaskItem.ts:32]: These should be private. Will update

**Pull request merged**